parse group config on groupable nodes

MichelleArk opened the pull request:
resolves https://github.com/dbt-labs/dbt-core/issues/6823

<!---
  Include the number of the issue addressed by this PR above if applicable.
  PRs for code changes without an associated issue *will not be merged*.
  See CONTRIBUTING.md for more information.
-->

### Description

<!---
  Describe the Pull Request here. Add any references and info to help reviewers
  understand your changes. Include any tradeoffs you considered.
-->

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

jtcohen6 commented:
> I'm wondering if it might not make more sense for generic tests to belong to the group of the node they're attached to.

Agreed! This also makes sense given `ref` restrictions we're putting in place in https://github.com/dbt-labs/dbt-core/issues/6826

MichelleArk commented:
@gshank - This is ready for another look! Main changes since the last review: 
* attached node lookup via `ref_lookup` and `disabled_lookup`: https://github.com/dbt-labs/dbt-core/pull/6965/commits/0712181c36e2aea2b4f4f12e6cf418ebf2b8395e
* added `group_map` to manifest for scheduling nodes in deleted groups for partial parsing: https://github.com/dbt-labs/dbt-core/pull/6965/commits/43b8d093f77d1110a62d37d18d5663afb2f3c8ce

gshank reviewed (approved):
Looks good!

I see what you meant about combining the ref_lookup and disabled. This actually feels like a somewhat better solution than in other places where it's too separate lookups.
